sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+"""
+    y_cable_base.py
+
+    Base class for implementing Y-Cable specific functionality in SONiC.
+    This is the base class of sonic_y_cable implentation. A vendor specific 
+    implementation of YCable needs to inherit from this class.
+    Classes derived from this class provide the ability to interact
+    with a vendor specific Y-Cable
+"""
+
+
+class YCableBase(object):
+
+    # definitions of targets for getting the cursor
+    # equalization parameters from the register spec
+    # the name of the target denotes which side cursor
+    # values will be retreived
+
+    TARGET_NIC = 0
+    TARGET_TOR1 = 1
+    TARGET_TOR2 = 2
+
+    # definitions of targets for getting the EYE/BER
+    # and initiating PRBS/Loopback on the Y cable
+    # the name of the target denotes which side values
+    # will be retreived/initiated
+
+    EYE_PRBS_TARGET_LOCAL = 0
+    EYE_PRBS_TARGET_TOR1 = 1
+    EYE_PRBS_TARGET_TOR2 = 2
+    EYE_PRBS_TARGET_NIC = 3
+
+    # definitions of switch counter types
+    # to be entered by the user in get_switch_count api
+    # for retreiving the counter values
+
+    SWITCH_COUNT_MANUAL = "manual"
+    SWITCH_COUNT_AUTO = "auto"
+
+    # switching modes inside muxcable
+    SWITCHING_MODE_MANUAL = 0
+    SWITCHING_MODE_AUTO = 1
+
+    # Valid return codes for upgrade firmware routine steps
+    FIRMWARE_DOWNLOAD_SUCCESS = 0
+    FIRMWARE_DOWNLOAD_FAILURE = 1
+    FIRMWARE_ACTIVATE_SUCCESS = 0
+    FIRMWARE_ACTIVATE_FAILURE = 1
+    FIRMWARE_ROLLBACK_SUCCESS = 0
+    FIRMWARE_ROLLBACK_FAILURE = 1
+
+    def __init__(self, port):
+        self.port = port
+
+    def toggle_mux_to_torA(self, port):
+        """
+        This API specifically does a hard switch toggle of the Y cable's MUX regardless of link state to
+        TOR A. This means if the Y cable is actively routing, the "check_active_linked_tor_side"
+        API will now return Tor A. It also implies that if the link is actively routing on this port, Y cable
+        MUX will start forwarding packets from TOR A to NIC, and drop packets from TOR B to NIC
+        regardless of previous forwarding state.
+
+        Args:
+            port:
+            an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a Boolean, true if the toggle succeeded and false if it did not succeed.
```

Comment:
Suggest capitalizing `True` and `False` to align with Python syntax. Please do this throughout the file.

Reply:
fixed